cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java

```diff
@@ -142,4 +165,132 @@ public void setMissingIncludeWarningsEnabled(boolean enabled){
+        while ((line = br.readLine()) != null) {
+          if (line.startsWith("  INCLUDE=")) { // handle environment includes 
+            String[] includes = line.split("=")[1].split(";");
+            for(String include : includes) {
+              if (!uniqueIncludes.contains(include)) {
+                uniqueIncludes.add(include);
+              }
+            }
+          }
+          
+          // get base path of project to make 
+          // Target "ClCompile" in file "C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.CppCommon.targets" from project "D:\Development\SonarQube\cxx\sonar-cxx\integration-tests\testdata\googletest_bullseye_vs_project\PathHandling.Test\PathHandling.Test.vcxproj" (target "_ClCompile" depends on it):
+          if (line.startsWith("Target \"ClCompile\" in file")) {
+            currentProjectPath = line.split("\" from project \"")[1].split("\\s+")[0].replace("\"", "");              
+          }
+          
+          if (line.contains("C:\\Program Files (x86)\\BullseyeCoverage\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);
+          }
+          
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 10.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 11.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 11.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);       
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);        
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);        
+          }  
+        }
+        br.close();
+      } catch (IOException ex) {
+        LOG.error("Cannot parse build log", ex);
+      }
+  }
+
+  private void parseVCppCompilerCLLine(String line, String projectPath) {
+    File file = new File(projectPath);
+    String project = file.getParent();
+
+    for (String includeElem : getMatches(Pattern.compile("/I\"(.*?)\""), line)) {
+      ParseInclude(includeElem, project);
+    }
+
+    for (String includeElem : getMatches(Pattern.compile("/I([^\\s\"]+) "),
+        line)) {
+      ParseInclude(includeElem, project);
+    }
+
+    for (String macroElem : getMatches(Pattern.compile("[/-]D\\s([^\\s]+)"),
```

Comment:
i think those we need to implement one by one. once issues are found. (if there are really issues). usually users when manually defining those they do not bother with them either, so if there is a concrete specification and a parsing error that occurs we can then fix it... but i would go and implement all kind of options at this moment since there are 2 many vc++ compilers to support.